### PR TITLE
Add force argument to /review command

### PR DESCRIPTION
**Pull request by prathamdby**
## Summary
- `/review force` cancels any queued or running review for the pull request and enqueues a replacement in one intake transaction.
- The replacement run resolves the latest head at execution time.
- A new `isReviewForceCommand` parser matches only `/review force` on the first non-empty line, case-sensitive.
- The ack executor processes `cancelProgress` before `progress`, so the new run's queued stub is the final comment state.
- New ack reply constants and `/help` text document the argument.
- README, operations, configuration, and queue-ops docs plus the site's llms knowledge describe the restart behavior.
- Tests cover the parser, force intake paths, ack ordering, and the cancel-then-requeue database flow.

## Details
- **Force intake:** `handleSlashReview` cancels active reviews with the issuer's login, releases their queue-slot jobs, then inserts and enqueues the new review.
- **Conflict path:** when the insert loses the uniqueness race after a force cancel, the ack still carries `cancelProgress` so the old run's check runs complete.
- **No active review:** force with nothing queued or running behaves like a plain `/review`.

### PR Type

Enhancement

### Description

- The change adds `isReviewForceCommand` in `parseSlashCommand.ts` to match `/review force` on the first non-empty line and two new ack bodies in `slashConstants.ts`.
- The change extends `handleSlashReview` in `slashIntake.ts` to cancel active reviews via `cancelActiveReviews` when `force` is true before creating a new work item.
- The change keeps intake atomic: it passes `cancelledWorkItemIds` to `releaseReviewQueueSlotInTx` and emits `agent_work_cancel_requested` and `agent_work_enqueued` with `force:true`.
- The change reorders `ackExecutor.ts` to run `cancelProgress` before `progress` so the cancelled check-run notice lands before the new queued progress stub.
- The change handles the uniqueness race: if creation loses, it still cleans check runs, replies with `SLASH_REVIEW_ALREADY_IN_PROGRESS_BODY`, and skips duplicate enqueue.
- The change enforces strict parsing: case-sensitive `force` with whitespace boundary prevents accidental restarts from `forcefully` or `Force`.
- The change updates docs and site knowledge (`README`, `operations.md`, `agent-work-ops.md`, `configuration.md`, `llmsKnowledge.ts`) to describe `/review force` behavior.

### Review map

1. <a href="https://github.com/prathamdby/pr-agent/pull/448/files#diff-fd02b6445593a4dd48d473f870fa98790b8c8260de189fc0a43f99adc1e30fc1"><code>src/agentWork/intake/slashIntake.ts</code></a>: Implements force cancel-then-enqueue transaction and reply selection
2. <a href="https://github.com/prathamdby/pr-agent/pull/448/files#diff-e6d6e214faadfaeb3ff1ee70099c91c9425a27b07c7c1a20561e70fe513ac648"><code>src/agentWork/executors/ackExecutor.ts</code></a>: Reorders cancel progress before queued stub publish
3. <a href="https://github.com/prathamdby/pr-agent/pull/448/files#diff-3c6699f90d0d67138dfa3d899f0efc4cbd1eec7261d8907c9461fd4a6bc89e94"><code>src/commands/parseSlashCommand.ts</code></a>: Adds strict first-line parser for /review force
4. <a href="https://github.com/prathamdby/pr-agent/pull/448/files#diff-21e478297ceba8cbb01eaa3c34d956b7a174aade2e204e2149bed46680abd4bc"><code>src/settings/slashConstants.ts</code></a>: Defines help text and ack bodies for force flow

### CODE REVIEW

File: test/ackExecutor.test.ts [hunk 1]
```diff
@@ -509,4 +513,52 @@ describe("executeAckJob", () => {
       }),
     );
   });
+
+  it("runs cancelProgress before progress when a force ack carries both", async () => {
+    const stub = renderReviewProgressComment({
+      mode: "review",
+      headSha: "sha-old",
+      source: "slash",
+      progressRevision: 1,
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply test/ackExecutor.test.ts:522]: Fixed in 688718a. New force-ack tests reject the cancel comment edit and the check-run cancellation separately; both assert the wi-new queued stub still publishes and the reply still sends.

File: test/slashIntake.test.ts [hunk 2]
```diff
@@ -730,4 +732,222 @@ describe("applySlashCommandIntake", () => {
       attributionPatch,
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply test/slashIntake.test.ts:732]: Fixed in 688718a. New intake test cancels a queued and a running row on /review force, pins cancelledWorkItemIds as [wi-running, wi-queued], and proves both ids reach the slot-release explicit cancel set by keeping the rows active so only that set can release the jobs.

File: test/parseSlashCommand.test.ts [hunk 3]
```diff
@@ -21,3 +21,25 @@ describe("parseSlashCommand", () => {
     expect(parseSlashCommand("")).toBe(null);
   });
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply test/parseSlashCommand.test.ts:22]: Fixed in 688718a. Added boundary cases: leading spaces or a tab before the slash reject, trailing space, tab, or newline after `force` matches. The double-space case was already pinned in the match test.

File: test/slashIntake.test.ts [hunk 4]
```diff
@@ -730,4 +732,222 @@ describe("applySlashCommandIntake", () => {
...
+      expect.objectContaining({
+        event: "agent_work_cancel_requested",
+        type: "review",
+        force: true,
+        cancelledByLogin: "alice",
+      }),
+    );
+    expect(intakeLog.getContext().events).toContainEqual(
+      expect.objectContaining({
+        event: "agent_work_enqueued",
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply test/slashIntake.test.ts:813]: Fixed in 688718a. Two tests pin empty and invalid (`@!invalid`) commenter logins sanitizing to "user" in cancelProgress.attribution, the cancelAttribution payload patch, and the agent_work_cancel_requested event.

File: src/agentWork/intake/slashIntake.ts [hunk 5]
```diff
@@ -292,8 +295,37 @@ async function handleSlashTriage(ctx: SlashIntakeContext): Promise<void> {
 
 async function handleSlashReview(ctx: SlashIntakeContext): Promise<void> {
   const resourceKey = prResourceKey(ctx.input.owner, ctx.input.repo, ctx.input.prNumber);
-  const alreadyInProgressBody =
-    "A `/review` run is already queued or in progress for this pull request.";
+  // `/review force`: cancel any queued/running review first so the fresh run
+  // below always starts on the latest head.
+  const force = isReviewForceCommand(ctx.input.body);
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply src/agentWork/intake/slashIntake.ts:300]: Fixed in 688718a. Unit test pins both cancel UPDATEs and the pg-boss lookup to the target PR's resource key; a new integration test seeds running reviews on two PRs and proves force cancels only the target PR's row and job while the sibling stays running.

File: src/agentWork/intake/slashIntake.ts [hunk 6]
```diff
@@ -292,8 +295,37 @@ async function handleSlashTriage(ctx: SlashIntakeContext): Promise<void> {
...
+  let cancelProgress: AckJobData["cancelProgress"];
+  if (force) {
+    const attribution = {
+      kind: "user" as const,
+      login: sanitizeGithubLogin(ctx.input.commenterLogin ?? ""),
+    };
+    const cancelled = await cancelActiveReviews(ctx.client, resourceKey, attribution);
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply src/agentWork/intake/slashIntake.ts:307]: Fixed in 688718a. The dual-row force test pins the running row as cancelProgress primary with ids [wi-running, wi-queued], and a race-loss variant asserts both ids survive on the ALREADY_IN_PROGRESS path.

File: src/agentWork/executors/ackExecutor.ts [hunk 7]
```diff
@@ -203,6 +203,27 @@ export async function executeAckJob(cfg: Config, pool: Pool, data: AckJobData):
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply src/agentWork/executors/ackExecutor.ts:204]: Fixed in 688718a. New test rejects findProgressComment and asserts ack_cancel_progress_failed is logged, the wi-new queued stub still publishes, and the reply still sends. Check cancellation is correctly never reached here (the throw happens first), so the test asserts it is not called.

File: src/agentWork/executors/ackExecutor.ts [hunk 8]
```diff
@@ -203,6 +203,27 @@ export async function executeAckJob(cfg: Config, pool: Pool, data: AckJobData):
 
   await prSurface.setAcknowledgementReaction(data.targets, GITHUB_REACTION_EYES);
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply src/agentWork/executors/ackExecutor.ts:205]: Fixed in 688718a. executeAckJob now hoists one prResourceKey(data.owner, data.repo, data.prNumber) from ../types.js and both manual template literals are gone.
